sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs

```diff
@@ -220,24 +220,28 @@ public void BlobUriBuilder_RegularUrl_CNAME()
         }
 
         [Test]
-        public void BlobUriBuilder_RegularUrl_MalformedSubdomain()
+        public void DataLakeUriBuilder_PartitionedAccountName()
         {
-            // core and blob swapped
-            var blobUriBuilder1 = new BlobUriBuilder(new Uri("https://account.core.blob.windows.net/container/blob"));
+            var blobUriBuilder = new BlobUriBuilder(new Uri("https://account.zone.blob.core.windows.net/share/dir"));
```

Comment:
If you are using a partitioned account, is "[account].[zone]" your literal account name?

Reply:
Now that I think about it, should we just create another value for zones now too? Instead of considering everything before the service subdomain the account name